workloads/files/workload-http-script-cm.yml

```diff
@@ -0,0 +1,1217 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: scale-ci-workload-script
+data:
+  run.sh: |
+    #!/bin/bash
+    set -e
+    # pbench Configuration
+    mkdir -p /var/lib/pbench-agent/tools-default/
+    echo "${USER_NAME:-default}:x:$(id -u):0:${USER_NAME:-default} user:${HOME}:/sbin/nologin" >> /etc/passwd
+    echo "" > /var/lib/pbench-agent/tools-default/oc
+    echo "workload" > /var/lib/pbench-agent/tools-default/label
+    if [[ -v ENABLE_PBENCH_AGENTS ]]; then
+      echo "" > /var/lib/pbench-agent/tools-default/disk
+      echo "" > /var/lib/pbench-agent/tools-default/iostat
+      echo "" > /var/lib/pbench-agent/tools-default/mpstat
+      echo "" > /var/lib/pbench-agent/tools-default/perf
+      echo "" > /var/lib/pbench-agent/tools-default/pidstat
+      master_nodes=`oc get nodes --show-labels -l pbench_agent=true,node-role.kubernetes.io/master= | grep master | awk '{print $1}'`
+      for node in $master_nodes; do
+        echo "master" > /var/lib/pbench-agent/tools-default/remote@$node
+      done
+      infra_nodes=`oc get nodes --show-labels -l pbench_agent=true,node-role.kubernetes.io/infra= | grep infra | awk '{print $1}'`
+      for node in $infra_nodes; do
+        echo "infra" > /var/lib/pbench-agent/tools-default/remote@$node
+      done
+      worker_nodes=`oc get nodes --show-labels -l pbench_agent=true,node-role.kubernetes.io/worker= | grep worker | awk '{print $1}'`
+      for node in $worker_nodes; do
+        echo "worker" > /var/lib/pbench-agent/tools-default/remote@$node
+      done
+    fi
+    source /opt/pbench-agent/profile
+    # End pbench Configuration
+
+    # Test Configuration
+    echo "$(date -u) Configuring HTTP test"
+    test_dir=/tmp/http-ci-tests
+    mkdir -p $test_dir/content/quickstarts/{nginx,stress}
+    install -m 644 /root/workload/server-*.yaml $test_dir/content/quickstarts/nginx
+    install -m 644 /root/workload/stress-*.yaml $test_dir/content/quickstarts/stress
+    install -m 755 /root/workload/*.sh $test_dir
+    #SERVER_RESULTS=${SERVER_RESULTS:-scp://root@${OCP_NODE_NAME}:/var/lib/pbench-agent}
+    SERVER_RESULTS=${SERVER_RESULTS:-kcp://scale-ci-tooling/${OCP_POD_NAME}:/var/lib/pbench-agent}
+    #echo "$(date -u) Test results will be copied to: $SERVER_RESULTS"
+    # End Test Configuration
+
+    # Start of Test Code
+    echo "$(date -u) Running HTTP test"
+    cd $test_dir
+    ./http-test.sh all
+    echo "$(date -u) Completed HTTP test"
+    # End of Test Code
+
+  ###################################################################################################
+  env.sh: |
```

Comment:
I looked in a bit more on how vars are moving here, currently AIUI we have:

User/Jenkins provides values via env vars -> Ansible reads env vars -> creates env configmap -> configmap is read by another env.sh script ( and establishes defaults) -> used in the script itself

I understand this design is to maintain "drop-in" compatibility with the upstream http tests, however is it at all possible to establish defaults at when Ansible reads the env vars thus eliminating the need for the env.sh script?  I am ok with it if there isn't in order to make "drop-in" with upstream easier and faster, so this is just an idea not a necessity.

Reply:
This really comes down to where you want the interface to be.  I've actually had the env variables read by defaulted by Ansible, but that was adding a lot of unnecessary code and not providing sane defaults at the level of the scripts themselves.  Also, as you mention, the scripts would no longer be self-contained unless you kept env.sh, which would lead to code duplication.  Neither approach is "great" in my view, but my approach at least preserves the "drop-in" functionality.  Will think about this more, but not sure I'll figure out a better solution.

Reply:
For current purposes, "drop-in" functionality is more important (at least in my opinion) so we can continue to think up ideas here and if anything seems to fit the bill for both of those, then we can propose a pr later for the change.